generators/server/templates/src/main/java/package/config/GatewaySwaggerResourcesProvider.java.ejs

```diff
@@ -0,0 +1,115 @@
+        this.routeLocator = routeLocator;
+        this.swaggerResources = swaggerResources;
+    }
+
+    @Override
+    public List<SwaggerResource> get() {
+        // Get gateway swagger resources and make their names more explicit
+        List<SwaggerResource> allSwaggerResources = swaggerResources.get();
+        allSwaggerResources.get(0).setName(String.format("gateway (%s)", allSwaggerResources.get(0).getName()));
+        allSwaggerResources.get(1).setName(String.format("gateway (%s)", allSwaggerResources.get(1).getName()));
+
+        List<Tuple2<Route, List<SwaggerResource>>> servicesRouteSwaggerResources = routeLocator
+            .getRoutes()
+            <% if (serviceDiscoveryType === 'consul') { %>
+            .filterWhen(
+                route -> {
+                    String routePredicate = route.getPredicate().toString();
+                    // Ignore the Consul server from the list as it doesn't expose a /swagger-resources endpoint
+                    return Mono.just(!routePredicate.contains("consul/consul"));
+                }
+            )
+            <% } %>
+            .flatMap(
+                route -> {
+                    // Retrieve the list of available OpenAPI resources for each service from their /swagger-resources endpoint
+                    WebClient serviceClient = WebClient.builder().baseUrl(route.getUri().toString()).build();
+                    Mono<List<SwaggerResource>> swaggerResources = serviceClient
+                        .get()
+                        .uri("/swagger-resources")
+                        .retrieve()
+                        .bodyToFlux(SwaggerResource.class)
+                        .onErrorResume(exception -> {
+                           exception.printStackTrace();
+                           return Mono.empty();
+                        })
+                        .collectList()
+                        .defaultIfEmpty(Collections.emptyList());
+                    return Mono.just(route).zipWith(swaggerResources);
+                }
+            )
+            .collectList()
+            .defaultIfEmpty(Collections.emptyList())
+            .subscribeOn(Schedulers.boundedElastic())
+            .toFuture()
+            .orTimeout(10, TimeUnit.SECONDS)
+            .join();
+
+        //Add the registered microservices swagger docs as additional swagger resources
+        List<SwaggerResource> servicesSwaggerResources = servicesRouteSwaggerResources
+            .stream()
+            .filter(tuple -> !tuple.getT2().isEmpty())
+            .map(
+                tuple -> {
+                    Route route = tuple.getT1();
+                    String routePredicate = route.getPredicate().toString();
+                    List<SwaggerResource> swaggerResources = tuple.getT2();
+                    List<SwaggerResource> swaggerResourcesFinal = new ArrayList<>();
+                    for (SwaggerResource swaggerResource : swaggerResources) {
```

Comment:
You can probably avoid creating variable `swaggerResources`, and thus rename your `swaggerResourcesFinal` variable to `swaggerResources`
```suggestion
                    for (SwaggerResource swaggerResource : tuple.getT2()) {
```